src/context/tasks-context.jsx

```diff
@@ -0,0 +1,35 @@
+import {
+  createContext,
+  useContext,
+  useEffect,
+  useReducer,
+  useState,
+} from "react";
+import { taskReducer } from "../utils";
+
+const tasksData = { tasks: [] };
+
+const TaskContext = createContext(null);
+
+const useTasks = () => useContext(TaskContext);
+
+const TaskProvider = ({ children }) => {
+  const [tasks, setTasks] = useState(
```

Comment:
This useState is not of use anymore here, as you are using useReducer here. You can handle all the things with the reducer only.

Reply:
Yes I can do that, useReducer will handle all the same functionalities, 
will update code accordingly